src/peft/tuners/lora/model.py

```diff
@@ -410,16 +418,18 @@ def add_weighted_adapter(
         combination_type = "linear" if len(adapters) == 1 else combination_type
 
         adapters_ranks = [self.peft_config[adapter].r for adapter in adapters]
-        if combination_type == "linear":
+        if combination_type in ("linear", "ties", "dare_ties", "dare_linear"):
             # all adapters ranks should be same, new rank is just this value
             if len(set(adapters_ranks)) != 1:
-                raise ValueError("All adapters must have the same r value when using `linear` combination_type")
+                raise ValueError(
+                    "All adapters must have the same r value when using combination_type linear, ties, dare_ties or dare_linear."
+                )
             new_rank = adapters_ranks[0]
         elif combination_type == "cat":
             # adapters ranks may be different, new rank is sum of all ranks
             # be careful, because output adapter rank may be really big if mixing a lot of adapters
             new_rank = sum(adapters_ranks)
-        elif combination_type == "svd":
+        elif "svd" in combination_type:
```

Comment:
I would prefer:

```suggestion
        elif (combination_type == "svd") or (combination_type == "ties_svd"):
```

This is to prevent potential bugs in the future where we add a new combination type with a strange name that has "svd" as a substring.

Reply:
but it is also meant for `dare_linear_svd`, `dare_ties_svd` too

Reply:
Oh I see. So maybe let's check all these options explicitly, or use `if combination_type.endswith("svd")`, which should be less error prone.